### ISSUE TITLE
iOS sample implementation flawed

### ISSUE THREAD

**Issue by prefectlyspelt** [paulpatarinski/Xamarin.Forms.Plugins#66]
Following the sample implementation in the sample app for ExtendedMaps on iOS.

using ExtendedMap.Forms.Plugin.iOS; 

results in the following error:

The type or namespace name `iOS' does not exist in the namespace `ExtendedMap.Forms.Plugin'. 

Searching through the libraries gives no resolution for the iOS side for 

ExtendedMapRenderer.Init ();

for Xamarin.Forms iOS projects.

**Comment by paulpatarinski** [comment]
Can you give me more details as to how you are getting this error? 

Is it when you try to run the Sample App in the root (the one containing all the plugins), the sample app part of the ExtendedMap or your own app?

**Comment by prefectlyspelt** [comment]
This is using the ExtendedMap in my own solution, as well as a similar issue in the root example. 

I'm using Xamarin Studio running on a Mac El Capitan, with the latest Xamarin Studio and getting the latest packages from NuGet for everything, including the latest Xamarin.Forms. 

Sample App (with all the plug ins, downloaded entire GitHub content, directly from GitHub via Zip):
TwoColumnCellRenderer fails on init.
 
ExtendedMap Sample App (ExtendedMap/ExtendedMap.sln, same source as above):
Not updating the packages in the solution, builds and runs fine.

Updated the packages in the solution, builds and runs fine.



My own solution:
I've created the most basic version of a solution which still shows the problem - https://www.dropbox.com/sh/ftg7clmjhsh4ioo/AADa9dpNfxvNbyspWhH7vmGMa?dl=0.


Steps were - 
create Xamarin.Forms Cross Platform PCL solution, only implement the iOS side,
add the latest packages from NuGet for: SVG, Extended Cells, Forms.Maps and Extended Maps to both the PCL project and the iOS project.
Ensure using and init statements match the working example in ExtendedMap.
Build - errors present as in screenshots.

**Comment by prefectlyspelt** [comment]
<img width="1279" alt="screen shot 2015-10-18 at 3 20 50 pm" src="https://github-cloud.s3.amazonaws.com/assets/15005254/10566420/6b5b1b4c-75ac-11e5-94ef-e730b0f0f222.png">

**Comment by prefectlyspelt** [comment]
<img width="1266" alt="extendedmapsample_appdelegate" src="https://github-cloud.s3.amazonaws.com/assets/15005254/10566439/c0441924-75ac-11e5-8131-15b4a7cc75be.png">

Working ExtendedMap sample

**Comment by prefectlyspelt** [comment]
<img width="1175" alt="screen shot 2015-10-19 at 9 46 22 am" src="https://cloud.githubusercontent.com/assets/15005254/10581053/5b8f22d6-7646-11e5-85b3-c3aea732055a.png">

Screenshot showing that ExtendedMap.Forms.Plugin.iOS is not available in Xam.Plugins.Forms.ExtendedMap as downloaded from NuGet using Xamarin Studio's package manager.

**Comment by robmpreston** [comment]
Yep, I had the same issue. I literally had to grab the iOS code and put the entire folder in my project to get it to work for now.

**Comment by paulpatarinski** [comment]
@robmpreston @prefectlyspelt Can you please retest with latest in master? I have upgraded to Forms 2.0 and fixed some reference issues

**Comment by prefectlyspelt** [comment]
Thanks Paul, I will test later today.

**Comment by prefectlyspelt** [comment]
Results are still the same with v1.5.0.1 - iOS is missing from ExtendedMap.Forms.Plugins as available from NuGet, but this is available using a version built from the source in GitHub.

<img width="409" alt="sampleapp_github" src="https://cloud.githubusercontent.com/assets/15005254/12106129/acf001b4-b321-11e5-891c-7bfd9806b147.png">
<img width="578" alt="v1 5 0 1_nuget_a" src="https://cloud.githubusercontent.com/assets/15005254/12106130/b4d5a51e-b321-11e5-95d3-1e74994a3cbf.png">
<img width="590" alt="v1 5 0 1_nuget" src="https://cloud.githubusercontent.com/assets/15005254/12106132/ba459b8a-b321-11e5-94c6-9b5b20e19041.png">

**Comment by Lukejkw** [comment]
@prefectlyspelt @paulpatarinski I am having the same issue. Did you manage to get around it?

**Comment by prefectlyspelt** [comment]
The only workaround I have found so far is to build from source rather than use the NuGet version.

**Comment by Lukejkw** [comment]
@prefectlyspelt Thanks. I also got it working by copying the iOS project locally. Works for playing around. 

Have you found a way to add click handlers to the `ExtendedPins` or is this functionality still not implemented?

I like Xamarin.Forms but the lack of a map control with custom pin icons while retaining the click event handler functionality has made it not viable for a couple of projects now. Have to go the Xamarin.Native route I guess.

**Comment by steevelefort** [comment]
Hi,
For those having the "The type or namespace name iOS does not exist in the namespace ExtendedMap.Forms.Plugin." problem.
You can add a reference to your iOS project and choose the "ExtendedMap.Forms.Plugin.iOS.dll" in the solution "Packages" folders.

**Comment by prefectlyspelt** [comment]
<img width="609" alt="screen shot 2016-02-05 at 11 59 12 am" src="https://cloud.githubusercontent.com/assets/15005254/12854571/086b6ca8-cc00-11e5-949c-7a657f2561be.png">

How do we add this reference? Do we need to do it by manually editing the references?

In Xamarin.Studio ExtendedMap.Forms.Plugin.iOS is not available under Edit References. Browsing for it in the packages folder _only_ shows the abstraction element (screenshot attached). I checked getting a completely fresh copy from NuGet and still only the abstraction element is available.

**Comment by steevelefort** [comment]
Yes I had to add the reference manually.
Then all works well for me.
